packages/twenty-server/src/engine/twenty-orm/utils/compute-permission-intersection.util.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  type ObjectsPermissionsDeprecated,
+  type RestrictedFieldPermissions,
+} from 'twenty-shared/types';
+
+export const computePermissionIntersection = (
+  permissionsArray: ObjectsPermissionsDeprecated[],
```

Comment:
Would it be costly to use the new ObjectPermissions instead of ObjectPermissionsDeprecated? It feels like we are creating quite a lot of debt here otherwise (canRead vs canReadObjectRecords etc)